libraries/chain/smt_evaluator.cpp

```diff
@@ -118,6 +115,15 @@ void smt_create_evaluator::do_apply( const smt_create_operation& o )
       token.control_account = o.control_account;
       token.market_maker.token_balance = asset( 0, token.liquid_symbol );
    });
+
+   auto nai_from_pool = asset_symbol_type::from_asset_num( o.symbol.get_stripped_precision_smt_num() );
+   const nai_pool_object* npo = _db.find< nai_pool_object, by_symbol >( nai_from_pool );
+   FC_ASSERT( npo, "NAI must have existed in the pool." );
+   _db.remove( *npo );
+   FC_ASSERT( !_db.asset_symbol_exists_in_nai_pool( o.symbol ), "Asset symbol should no longer be in NAI pool." );
```

Comment:
This seems like an assertion that should live in a test case rather than the evaluator.

Reply:
Also, at this point in the method, you've looked up this symbol 3 times, when you only need one lookup.